frontend/src/app/modules/project/components/project-window/project-window.component.ts

```diff
@@ -0,0 +1,284 @@
+        else if(control.hasError('max')) {
+            errorMessage = `Quantity must be less than ${control.errors.max.max}!`;
+        }
+        else if (control.hasError('minlength')) {
+            errorMessage = `The length must be at least ${control.errors.minlength.requiredLength} letters!`;
+        }
+        else if (control.hasError('maxlength')) {
+            errorMessage = `The length should be no more than ${control.errors.maxlength.requiredLength} letters!`;
+        }
+        else {
+            errorMessage = 'validation error';
+        }
+        return errorMessage;
+    }
+
+    public close() {
+        this.ref.close();
+    }
+        
+    private InitializeProject() {
+        this.projectForm.get('name').setValue(this.projectUpdateStartState.name);
+        this.projectForm.get('description').setValue(this.projectUpdateStartState.description);
+        this.projectForm.get('countOfSavedBuilds').setValue(this.projectUpdateStartState.countOfSaveBuilds);
+        this.projectForm.get('countOfBuildAttempts').setValue(this.projectUpdateStartState.countOfBuildAttempts);
+        this.projectForm.get('color').setValue(this.projectUpdateStartState.color);
+    }
+
+    private SetProjectObjectsFromResponse(resp: HttpResponse<ProjectInfoDTO>): void {
+        const res = resp.body;
+        this.projectUpdateStartState = resp.body;
+        this.projectUpdate.id = res.id;
+        this.projectUpdate.name = res.name;
+        this.projectUpdate.description = res.description;
+        this.projectUpdate.countOfSaveBuilds = res.countOfSaveBuilds;
+        this.projectUpdate.countOfBuildAttempts = res.countOfBuildAttempts;
+        this.projectUpdate.color = res.color;
+        this.projectUpdate.accessModifier = res.accessModifier;
+    }
+
+    private getValuesForProjectUpdate() {
+        this.projectUpdate.name = this.projectForm.get('name').value;
+        this.projectUpdate.description = this.projectForm.get('description').value;
+        this.projectUpdate.countOfSaveBuilds = this.projectForm.get('countOfSavedBuilds').value;
+        this.projectUpdate.countOfBuildAttempts = this.projectForm.get('countOfBuildAttempts').value;
+        this.projectUpdate.color = this.projectForm.get('color').value;
+    }
+
+    private getValuesForProjectCreate() {
+        this.projectCreate.name = this.projectForm.get('name').value;
+        this.projectCreate.color = this.projectForm.get('color').value;
+        this.projectCreate.compilerType = this.projectForm.get('compilerType').value;
+        this.projectCreate.countOfBuildAttempts = this.projectForm.get('countOfBuildAttempts').value;
+        this.projectCreate.countOfSaveBuilds = this.projectForm.get('countOfSavedBuilds').value;
+        this.projectCreate.description = this.projectForm.get('description').value;
+        this.projectCreate.language = this.projectForm.get('language').value;
+        this.projectCreate.projectType = this.projectForm.get('projectType').value;
```

Comment:
better use `this.projectForm.setValue()`

Reply:
Thank you)